plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleClient.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.oracle;
+
+import io.trino.plugin.jdbc.BaseJdbcConfig;
+import io.trino.plugin.jdbc.DefaultQueryBuilder;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.WriteFunction;
+import io.trino.plugin.jdbc.WriteMapping;
+import io.trino.plugin.jdbc.mapping.DefaultIdentifierMapping;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.Type;
+import io.trino.testing.TestingConnectorSession;
+import oracle.jdbc.OracleTypes;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import static com.google.common.reflect.Reflection.newProxy;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.CharType.createCharType;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_SECONDS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestOracleClient
+{
+    private static final JdbcClient CLIENT = new OracleClient(
+            new BaseJdbcConfig(),
+            new OracleConfig(),
+            session -> {
+                throw new UnsupportedOperationException();
+            },
+            new DefaultQueryBuilder(),
+            new DefaultIdentifierMapping());
+
+    private static final ConnectorSession SESSION = TestingConnectorSession.SESSION;
+
+    @Test(dataProvider = "writeMappingsProvider")
+    public void testTypedNullWriteMapping(Type type, String bindExpression, int nullJdbcType)
+            throws SQLException
+    {
+        WriteMapping writeMapping = CLIENT.toWriteMapping(SESSION, type);
+        assertThat(writeMapping.getWriteFunction()).isNotNull();
+        WriteFunction writeFunction = writeMapping.getWriteFunction();
+
+        assertThat(writeFunction.getBindExpression()).isEqualTo(bindExpression);
+
+        PreparedStatement statementProxy = newProxy(PreparedStatement.class, (proxy, method, args) -> {
+            assertThat(method.getName()).isEqualTo("setNull");
```

Comment:
add a code-level comment why it matters.
copy the info from commit message

(think that the test class could be moved, refactored, split, extracted and the original code message may get lost; now, should a test fail, an engineer will want to know why it matters at all)

Reply:
Comment added.